### ISSUE TITLE
Support `iNZightTools`'s survey design file format?

### ISSUE THREAD

**Issue by gergness** [gergness/srvyr#128]
It feels like such a good idea to have a file format with the survey design information in it, I'd love to help try and make this a thing. I think the best course of action would be to re-export `iNZightTools::import_survey`, but maybe there should also be an `as_survey()` wrapper too. Any thoughts @tmelliott?

**Comment by tmelliott** [comment]
Thanks @gergness - iNZightTools is an absolute mess of a package. I have plans to move as many Depends to Suggests as possible to make it a lighter dependency, but *after* (or while) I've gone through and re-factored many of the functions to be more CLI friendly (most have bad function/argument names only ever intended for the iNZight GUI, but our vision has since changed). 

I'd definitely be interested in some collaboration if you have other/better/alternative ideas (e.g., implementing `as_survey()` which would be both very useful and simple :man_facepalming:). `import_survey()` is very much a work-in-progress and could be a lot better, and help designing/testing would be very much appreciated!

**Comment by gergness** [comment]
Ha, the `as_survey` I was thinking about implementing would be to add a `file=` argument to them `as_survey` methods in  srvyr, so that it could read a survey spec from disk, so that's something I would need to do.

Unfortunately, I still haven't had time to look closely enough to give you useful feedback for designing and testing. My instincts are to narrow scope to be just about the survey design definitions (and not data import), but that might just be my self-interest because it would mean fewer (suggested) dependencies for srvyr. I also may have missed it, but the other thing I was missing is a document that defines what goes into a survey.design file.

**Comment by tmelliott** [comment]
I am considering splitting out a separate package purely for working with/distributing surveys and decoupling it from iNZight entirely. Something like `surveyspec::read()`... thoughts?

**Comment by gergness** [comment]
Yeah, I think that makes a lot of sense! From srvyr's perspective, it would be nice if it the data import stuff was kept separate from that too, but that may not make sense from your perspective.

**Comment by gergness** [comment]
(I'm 1 week into a month of parental leave with a 3 month old, so am in a weird spot where I crave the intellectual stimulation of a side project like this, but am especially flaky)

**Comment by tmelliott** [comment]
Cool - maybe I'll try start a repo in the coming days to get the ball rolling on that while you've got some 'free' time.. :baby:

**Comment by tmelliott** [comment]
FYI [surveyspec](github.com/tmelliott/surveyspec) is now a thing.
